test/hotspot/gtest/riscv/test_assembler_riscv.cpp

```diff
@@ -322,151 +326,374 @@ TEST_VM(RiscV, cmpxchg_int16_lr_sc) {
   UseZacas = zacas;
 }
 
+TEST_VM(RiscV, cmpxchg_int16_maybe_zacas) {
+  if (UseZacas) {
+    run_narrow_cmpxchg_tests<int16_t, Assembler::int16>();
+  }
+}
+
 TEST_VM(RiscV, cmpxchg_int8_lr_sc) {
   bool zacas = UseZacas;
   UseZacas = false;
   run_narrow_cmpxchg_tests<int8_t, Assembler::int8>();
   UseZacas = zacas;
 }
 
-TEST_VM(RiscV, cmpxchg_int16_maybe_zacas) {
+TEST_VM(RiscV, cmpxchg_int8_maybe_zacas) {
   if (UseZacas) {
-    run_narrow_cmpxchg_tests<int16_t, Assembler::int16>();
+    run_narrow_cmpxchg_tests<int8_t, Assembler::int8>();
   }
 }
 
-TEST_VM(RiscV, cmpxchg_int8_maybe_zacas) {
+template <typename TESTSIZE>
+TESTSIZE next_count(TESTSIZE now, TESTSIZE add) {
+  if ((std::numeric_limits<TESTSIZE>::max() - add) >= now) {
+    return now + add;
+  }
+  TESTSIZE diff = std::numeric_limits<TESTSIZE>::max() - now;
+  add -= diff;
+  return std::numeric_limits<TESTSIZE>::min() + add;
```

Comment:
`next_count(127, 1)` returns -127. Shouldn't we get -128 after the wrap around?

Reply:
Yes, you are correct. Forgot to subtract one for the wrap around count.